corehq/ex-submodules/casexml/apps/case/mock/case_block.py

```diff
@@ -2,145 +2,167 @@
 from datetime import datetime, date
 import numbers
 from xml.etree import ElementTree
-from casexml.apps.case.xml import NS_VERSION_MAP, V2
+from casexml.apps.case.xml import V2_NAMESPACE
 from dimagi.utils.parsing import json_format_datetime
+from collections import namedtuple
+from functools import partial
 
+# relationship = "child" for index to a parent case (default)
+# relationship = "extension" for index to a host case
+IndexAttrs = namedtuple('IndexAttrs', ['case_type', 'case_id', 'relationship'])
+ChildIndexAttrs = partial(IndexAttrs, relationship='child')
 
-class CaseBlock(dict):
+
+class CaseBlock(object):
+    """
+    see https://github.com/dimagi/commcare/wiki/casexml20 for spec
+    """
     undefined = object()
 
     def __init__(self, case_id, date_modified=None, user_id=undefined,
                  owner_id=undefined, external_id=undefined, case_type=undefined,
                  case_name=undefined, create=False, date_opened=undefined, update=None,
-                 close=False, index=None, strict=True):
-        """
-        see https://github.com/dimagi/commcare/wiki/casexml20 for spec
-        """
-        super(CaseBlock, self).__init__()
-        self._id = case_id
+                 close=False, index=None):
+        # todo: can we use None instead of CaseBlock.undefined, throughout?
```

Comment:
I think that should be fine unless there are properties that you want to have the ability to unset in which case you'd want to be able to specify `None` to unset the property or `undefined` to not change it. The only property I can think of that this might apply to is `external_id`